dot/network/state.go

```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/ChainSafe/gossamer/lib/common"
 )
 
-//go:generate mockery --name BlockState --structname MockBlockState --case underscore --inpackage
+//go:generate mockgen -destination=mock_block_state.go -package $GOPACKAGE . BlockState
```

Comment:
Write it to a file with a `_test.go` suffix so it doesn't pollute the Go API perhaps? Same for below if you can / it's not imported by other packages.